google/cloud/firestore_v1/_helpers.py

```diff
@@ -219,6 +232,36 @@ def encode_dict(values_dict) -> dict:
     return {key: encode_value(value) for key, value in values_dict.items()}
 
 
+def document_snapshot_to_protobuf(snapshot: "google.cloud.firestore_v1.base_document.DocumentSnapshot") -> Optional["google.cloud.firestore_v1.types.Document"]:  # type: ignore
+    from google.cloud.firestore_v1.types import Document
+
+    if not snapshot.exists:
+        return None
+
+    return Document(
+        name=f"{snapshot._client._database_string}/{snapshot.reference.path}",
+        fields=encode_dict(snapshot._data),
+        create_time=snapshot.create_time,
+        update_time=snapshot.update_time,
+    )
+
+
+def parse_reference_value(reference_value: str) -> Dict[str, str]:
```

Comment:
This looks to be extracting code from `reference_value_to_document` but that code hasn't been replaced to use this. Is this different from that code?  Perhaps this should give names to all of the splits? since it seems like for the other method we really only use the final split.

Is 'reference_value' a good name for this? it might be alright, though I think of these as 'document_path'. Curious about your thoughts on some of the terminology we use.

Reply:
Collapsed this functionality by introducing a utility class and deleting my original function. Ideally that utility class would be immutable, but wiring all of that up felt like a bit of overkill.